org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend

```diff
@@ -15,7 +14,7 @@ class GeneratedJavaFileAccess extends JavaFileAccess {
 	CharSequence typeComment
 	
 	@Accessors
-	val List<IClassAnnotation> annotations = newArrayList
+	val Set<IClassAnnotation> annotations = newLinkedHashSet
```

Comment:
Same as above: it's not necessary to change this to a set if you do `.toSet` in `getClassAnnotations()`